### ISSUE TITLE
Stable branch stalls at boot 

### ISSUE THREAD

**Issue by albertdb** [steigr/coreos#2]
OS: Arch Linux
Kernel: 3.18.25

If doesn't reach bash login, bash shell (or whatever it reaches, it's the first time I try it). A `docker exec <id> bash`, ran after the boot process apparently hangs, shows that it hasn't finished boot process, so something is going wrong.

```
# docker run --privileged --rm --tty --interactive steigr/coreos
Unable to find image 'steigr/coreos:latest' locally
latest: Pulling from steigr/coreos

7a84e6f5dd0c: Pull complete 
c80fa55164e7: Pull complete 
7f87dffd2ad4: Pull complete 
1a52c4aef59a: Pull complete 
871d2cf68b7e: Pull complete 
Digest: sha256:a4d09ce71b7ef5bb702799d9c1544b41fd69ce19ef41aa65a0fe77e59ec05de2
Status: Downloaded newer image for steigr/coreos:latest
systemd 222 running in system mode. (-PAM +AUDIT -SELINUX +IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT -GNUTLS -ACL +XZ -LZ4 +SECCOMP +BLKID -ELFUTILS +KMOD -IDN)
Detected virtualization docker.
Detected architecture x86-64.
Running with unpopulated /etc.

Welcome to CoreOS 766.5.0!

Set hostname to <686f23124ea0>.
Initializing machine ID from random generator.
Populated /etc with preset unit settings.
Failed to install release agent, ignoring: No such file or directory
etc-hosts.mount: Unit is bound to inactive unit dev-sdb1.device. Stopping, too.
etc-hostname.mount: Unit is bound to inactive unit dev-sdb1.device. Stopping, too.
etc.mount: Unit is bound to inactive unit dev-sdb1.device. Stopping, too.
mnt.mount: Unit is bound to inactive unit dev-sdb1.device. Stopping, too.
var.mount: Unit is bound to inactive unit dev-sdb1.device. Stopping, too.
etc-resolv.conf.mount: Unit is bound to inactive unit dev-sdb1.device. Stopping, too.
root.mount: Unit is bound to inactive unit dev-sdb1.device. Stopping, too.
systemd-udevd.service: Cannot add dependency job, ignoring: Unit systemd-udevd.service is masked.
dev-hugepages.mount: Cannot add dependency job, ignoring: Unit dev-hugepages.mount is masked.
systemd-machine-id-commit.service: Cannot add dependency job, ignoring: Unit systemd-machine-id-commit.service is masked.
systemd-udev-settle.service: Cannot add dependency job, ignoring: Unit systemd-udev-settle.service is masked.
usr-share-oem.mount: Cannot add dependency job, ignoring: Unit usr-share-oem.mount is masked.
systemd-udev-settle.service: Cannot add dependency job, ignoring: Unit systemd-udev-settle.service is masked.
tmp.mount: Cannot add dependency job, ignoring: Unit tmp.mount is masked.
proc-sys-fs-binfmt_misc.automount: Cannot add dependency job, ignoring: Unit proc-sys-fs-binfmt_misc.automount is masked.
systemd-udev-trigger.service: Cannot add dependency job, ignoring: Unit systemd-udev-trigger.service is masked.
systemd-udevd-control.socket: Cannot add dependency job, ignoring: Unit systemd-udevd-control.socket is masked.
systemd-udevd-kernel.socket: Cannot add dependency job, ignoring: Unit systemd-udevd-kernel.socket is masked.
audit-rules.service: Cannot add dependency job, ignoring: Unit audit-rules.service is masked.
[  OK  ] Reached target Swap.
[  OK  ] Created slice Root Slice.
[  OK  ] Listening on networkd rtnetlink socket.
[  OK  ] Listening on /dev/initctl Compatibility Named Pipe.
[  OK  ] Listening on Journal Socket (/dev/log).
[  OK  ] Listening on Journal Socket.
[  OK  ] Created slice System Slice.
         Starting Activation of LVM2 logical volumes...
         Mounting Debug File System...
         Starting Rebuild Hardware Database...
         Starting Setup Virtual Console...
[  OK  ] Created slice system-system\x2dcloudinit.slice.
         Starting Rebuild Dynamic Linker Cache...
[  OK  ] Created slice system-getty.slice.
         Starting Create /etc/nsswitch.conf...
         Starting Load/Save Random Seed...
         Starting Create System Users...
         Starting Apply Kernel Variables...
         Mounting FUSE Control File System...
         Mounting Configuration File System...
         Starting Journal Service...
[  OK  ] Created slice User and Session Slice.
[  OK  ] Reached target Slices.
[  OK  ] Mounted Debug File System.
[  OK  ] Mounted Configuration File System.
[  OK  ] Started Setup Virtual Console.
[  OK  ] Started Create /etc/nsswitch.conf.
[  OK  ] Started Load/Save Random Seed.
[  OK  ] Started Create System Users.
[  OK  ] Mounted FUSE Control File System.
         Starting Create Static Device Nodes in /dev...
[  OK  ] Started Apply Kernel Variables.
[  OK  ] Started Create Static Device Nodes in /dev.
[  OK  ] Reached target Local File Systems (Pre).
[  OK  ] Started Journal Service.
         Starting Flush Journal to Persistent Storage...
[  OK  ] Started Activation of LVM2 logical volumes.
[  OK  ] Started Rebuild Hardware Database.
[  OK  ] Started Rebuild Dynamic Linker Cache.
[  OK  ] Reached target Encrypted Volumes.
         Starting Activation of LVM2 logical volumes...
[  OK  ] Started Flush Journal to Persistent Storage.
[  OK  ] Started Activation of LVM2 logical volumes.
         Starting Activation of LVM2 logical volumes...
[  OK  ] Reached target Local File Systems.
         Starting Rebuild Journal Catalog...
         Starting Create missing system files...
         Starting Create Volatile Files and Directories...
[  OK  ] Started Rebuild Journal Catalog.
[  OK  ] Started Create missing system files.
[  OK  ] Started Create Volatile Files and Directories.
[  OK  ] Reached target System Time Synchronized.
         Starting Update UTMP about System Boot/Shutdown...
         Starting Clean up broken links in /etc/ssl/certs...
         Starting Update is Completed...
[  OK  ] Started Update is Completed.
[  OK  ] Started Update UTMP about System Boot/Shutdown.
[  OK  ] Started Clean up broken links in /etc/ssl/certs.
[  OK  ] Reached target System Initialization.
[  OK  ] Listening on D-Bus System Message Bus Socket.
[  OK  ] Listening on OpenSSH Server Socket.
[  OK  ] Reached target Paths.
[  OK  ] Reached target Timers.
[  OK  ] Listening on rkt metadata service socket.
[  OK  ] Listening on Docker Socket for the API.
         Starting Cleanup of Temporary Directories...
[  OK  ] Reached target Sockets.
[  OK  ] Reached target Basic System.
         Starting Garbage Collection for rkt...
[  OK  ] Started Console Getty.
         Starting Console Getty...
[  OK  ] Started Getty on tty1.
         Starting Getty on tty1...
[  OK  ] Reached target Login Prompts.
         Starting Generate sshd host keys...
         Starting Install an ssh key from /proc/cmdline...
[  OK  ] Started D-Bus System Message Bus.
         Starting D-Bus System Message Bus...
         Starting Network Service...
         Starting Login Service...
         Starting Generate /run/coreos/motd...
[  OK  ] Reached target Load system-provided cloud configs.
[  OK  ] Reached target Load user-provided cloud configs.
[  OK  ] Started Activation of LVM2 logical volumes.
[  OK  ] Started Cleanup of Temporary Directories.
[  OK  ] Started Garbage Collection for rkt.
[  OK  ] Started Install an ssh key from /proc/cmdline.
[  OK  ] Reached target Remote File Systems.
[  OK  ] Started Login Service.
[  OK  ] Started Network Service.
[  OK  ] Reached target Network.
         Starting Network Name Resolution...
[  OK  ] Started Generate /run/coreos/motd.
[  OK  ] Started Network Name Resolution.
[  OK  ] Started Generate sshd host keys.
         Starting Generate /run/issue...
[  OK  ] Started Generate /run/issue.
[  OK  ] Reached target Multi-User System.

```

**Comment by steigr** [comment]
Intentionally CoreOS read's /proc/cmdline and create a drop-in for getty to drop you if there's "cores.autologin" in /proc/cmdline. There are at least 3 things to consider:

At first, /proc/cmdline belongs to your hosts environment and must not be parsed. My current local builder changes it to /etc/cmdline.

The seconds thing is, that your terminal is neither /dev/ttyS* nor /dev/tty*. So system-console.service has to be changed somehow.

Last but not least there should be something to signal if autologin should happen or not. "--tty" and "--interactive" are those switches in the first place, but i don't know how to reliable check from inside the container if these are set.

**Comment by albertdb** [comment]
Thanks for answering.

It seems that the comment got lost when closing the issue.

It is working. I noticed when shutting down the host and it changed to tty1, where I saw the CoreOS command prompt. It seems that CoreOS container replaces tty1 to tty5.

The only thing that I noticed that doesn't work is the shutdown procedure. I haven't managed to stop or kill CoreOS container once I start it without having to reboot host machine.
